pkg/cli/admin/upgrade/status/workerpool.go

```diff
@@ -158,14 +158,17 @@ func assessNodesStatus(cv *configv1.ClusterVersion, pool mcfgv1.MachineConfigPoo
 		isUnavailable := lns.IsUnavailable(&pool)
 
 		isDegraded := isNodeDegraded(node)
+		annotationMatched := node.Annotations[mco.CurrentMachineConfigAnnotationKey] == desiredConfig
+		desiredAnnotationUpdated := desiredConfig == pool.Spec.Configuration.Name
 		isUpdated := foundCurrent && isLatestUpdateHistoryVersionEqualTo(cv.Status.History, currentVersion) &&
 			// The following condition is to handle the multi-arch migration because the version number stays the same there
-			(!ok || node.Annotations[mco.CurrentMachineConfigAnnotationKey] == desiredConfig)
+			(!multiArchMigration || !ok || (annotationMatched && desiredAnnotationUpdated))
 
 		// foundCurrent makes sure we don't blip phase "updating" for nodes that we are not sure
 		// of their actual phase, even though the conservative assumption is that the node is
 		// at least updating or is updated.
-		isUpdating := !isUpdated && foundCurrent && foundDesired && isLatestUpdateHistoryVersionEqualTo(cv.Status.History, desiredVersion)
+		isUpdating := !isUpdated && foundCurrent && foundDesired && isLatestUpdateHistoryVersionEqualTo(cv.Status.History, desiredVersion) &&
+			(!annotationMatched && desiredAnnotationUpdated)
```

Comment:
My thinking is `(!annotationMatched && desiredAnnotationUpdated)` is implicitly `true` (provided `!isUpdated && foundCurrent && foundDesired && isLatestUpdateHistoryVersionEqualTo(cv.Status.History, desiredVersion)` is true) for an updating node during a normal version update. (It is necessary only for multi-arch migration tho.)
Otherwise, one of the following conditions is true:
- `node.Annotations[mco.CurrentMachineConfigAnnotationKey]==node.Annotations[mco.DesiredMachineConfigAnnotationKey]` which would contradicts to `!isUpdated`.
- `desiredConfig != pool.Spec.Configuration.Name`. I bet MCO always sets those two up during update even if it is an optional field.